New nuclei crashes once again

### Nuclei version: v3.2.8
<!-- You can find current version of nuclei with "nuclei -version" -->
<!-- We only accept issues that are reproducible on the latest version of nuclei. -->
<!-- You can find the latest version of project at https://github.com/projectdiscovery/nuclei/releases/ -->

### Current Behavior:
On Medium, High, Critical im getting following error 
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1d48746]

goroutine 1466 [running]:
github.com/projectdiscovery/nuclei/v3/pkg/protocols/http/httpclientpool.SendRawRequest(0x27fdac0?, 0xc00d349328)
        github.com/projectdiscovery/nuclei/v3/pkg/protocols/http/httpclientpool/options.go:43 +0x126
github.com/projectdiscovery/nuclei/v3/pkg/protocols/http.(*Request).executeRequest(0xc000b914a0, 0xc00806d8c0, 0xc003303080, 0x0, 0x0, 0xc00d349970?, 0x0?)
        github.com/projectdiscovery/nuclei/v3/pkg/protocols/http/request.go:740 +0x3b59
github.com/projectdiscovery/nuclei/v3/pkg/protocols/http.(*Request).ExecuteWithResults.func1({0xc008d2b240, 0x3b}, 0xc00d349a98?, 0x418b08?)
        github.com/projectdiscovery/nuclei/v3/pkg/protocols/http/request.go:516 +0x5b4
github.com/projectdiscovery/nuclei/v3/pkg/protocols/http.(*Request).ExecuteWithResults(0xc000b914a0, 0xc00806d8c0, 0xc00bc3c7b0, 0x0, 0xc009d93f08)
        github.com/projectdiscovery/nuclei/v3/pkg/protocols/http/request.go:588 +0x3c8
github.com/projectdiscovery/nuclei/v3/pkg/tmplexec/multiproto.(*MultiProtocol).ExecuteWithResults(0xc00663ef60, 0xc008a812c0)
        github.com/projectdiscovery/nuclei/v3/pkg/tmplexec/multiproto/multi.go:114 +0x217
github.com/projectdiscovery/nuclei/v3/pkg/tmplexec.(*TemplateExecuter).Execute(0xc00102fd00, 0xc008a812c0)
        github.com/projectdiscovery/nuclei/v3/pkg/tmplexec/exec.go:197 +0x43c
github.com/projectdiscovery/nuclei/v3/pkg/core.(*Engine).executeTemplateWithTargets.func2.1(0x11?, 0x0?, 0xc005b11a00)
        github.com/projectdiscovery/nuclei/v3/pkg/core/executors.go:139 +0x203
created by github.com/projectdiscovery/nuclei/v3/pkg/core.(*Engine).executeTemplateWithTargets.func2 in goroutine 1451
        github.com/projectdiscovery/nuclei/v3/pkg/core/executors.go:115 +0x511

```

![3dfe2b05f81bc10bc0c9c628870cff5f](https://github.com/projectdiscovery/nuclei/assets/10222100/c61d812c-28b8-4202-b7dc-c784f36282d1)





### Expected Behavior:
To finish scan without issues

### Steps To Reproduce:

`cat webs_nuclei.txt 2>/dev/null | nuclei -silent -retries 2 -etags openssh -s medium,high,critical -nh -rl 150 -o nuclei_output.txt`

--- a/pkg/protocols/http/httpclientpool/options.go
+++ b/pkg/protocols/http/httpclientpool/options.go
@@ -7,6 +7,7 @@ import (
 	"time"
 
 	"github.com/projectdiscovery/rawhttp"
+	urlutil "github.com/projectdiscovery/utils/url"
 )
 
 // WithCustomTimeout is a configuration for custom timeout
@@ -39,7 +40,27 @@ func SendRawRequest(client *rawhttp.Client, opts *RawHttpRequestOpts) (*http.Res
 			// this error is caused when rawhttp client sends a corrupted or malformed request packet to server
 			// some servers may attempt gracefully shutdown but most will just abruptly close the connection which results
 			// in a connection reset by peer error and this can be safely assumed as 400 Bad Request in terms of normal http flow
-			req, _ := http.NewRequest(opts.Method, opts.URL, opts.Body)
+			req, reqErr := http.NewRequest(opts.Method, opts.URL, opts.Body)
+			if reqErr != nil {
+				// failed to build new request mostly because of invalid url or body
+				// try again or else return urlErr
+				parsed, urlErr := urlutil.ParseAbsoluteURL(opts.URL, true)
+				if urlErr != nil {
+					return nil, err
+				}
+				req, reqErr = http.NewRequest(opts.Method, parsed.Host, opts.Body)
+				if reqErr != nil {
+					return nil, err
+				}
+				req.URL = parsed.URL
+				req.Header = opts.Headers
+			}
+
+			// if req is still nil, return error
+			if req == nil {
+				return nil, err
+			}
+
 			req.Header = opts.Headers
 			resp = &http.Response{
 				Request:    req,

